homeassistant/components/github/icons.json

```diff
@@ -0,0 +1,42 @@
+{
```

Comment:
Is there really no way of defining a default icon for the entire integration/platforms?

Reply:
Nope

Reply:
I did replace some Github icons with some other fitting icons

Reply:
I do not like this move.
I think we should add a default icon for integration and/or platform.
If that can not be done, a test in this integration should be added to ensure that no entities have `None` as the icon.

Reply:
Entities will have None as the icon after this move because this move moves it out of the state machine.

For all the integrations in A-G I found only 1 integration that would benefit from having a fallback icon on integration level and that's Enphase envoy.

Reply:
It can be tested like this:

```python
async def test_sensor_icons(
    hass: HomeAssistant,
    init_integration: MockConfigEntry,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test to ensure that all sensor entities have an icon definition."""
    entities = er.async_entries_for_config_entry(
        entity_registry,
        config_entry_id=init_integration.entry_id,
    )

    icons = await icon.async_get_icons(hass, "entity", integrations=["github"])
    for entity in entities:
        assert entity.translation_key is not None
        assert icons["github"]["sensor"][entity.translation_key] is not None
```